src/libs/bin_decoder/lv_bin_decoder.c

```diff
@@ -585,7 +588,8 @@ static lv_result_t decode_indexed(lv_image_decoder_t * decoder, lv_image_decoder
         decoder_data->palette = (void *)palette; /*Need to free when decoder closes*/
 
 #if LV_BIN_DECODER_RAM_LOAD
-        draw_buf_indexed = lv_draw_buf_create(dsc->header.w, dsc->header.h, cf, dsc->header.stride);
+        draw_buf_indexed = lv_draw_buf_create_user(image_cache_draw_buf_handlers, dsc->header.w, dsc->header.h, cf,
```

Comment:
It may be better we define a common function in lv_image_decoder.c called `lv_image_decoder_draw_buf_create` or similar. Call it instead in all decoders.

Same for `lv_draw_buf_dup`.

We also need a method to detect if user has misused the draw buffer handlers somehow.

Reply:
> It may be better we define a common function in lv_image_decoder.c called `lv_image_decoder_draw_buf_create` or similar. Call it instead in all decoders.
> 
> Same for `lv_draw_buf_dup`.
> 
> We also need a method to detect if user has misused the draw buffer handlers somehow.

Great idea!

Let's make it happen in the near future.